specification/http/1.0/openapi.json

```diff
@@ -0,0 +1,456 @@
+                            }
+                          }
+                        },
+                        "additionalProperties": false
+                      }
+                    }
+                  },
+                  "required": ["id"],
+                  "additionalProperties": false
+                },
+                "secrets": {
+                  "type": "array",
+                  "items": {
+                    "type": "object",
+                    "properties": {
+                      "id": {
+                        "type": "string",
+                        "description": "The secret's unique identifier."
+                      }
+                    },
+                    "required": ["id"],
+                    "additionalProperties": false
+                  }
+                }
+              },
+              "additionalProperties": false
+            }
+          },
+          "required": ["id", "name", "description", "toolkit"],
+          "additionalProperties": false
+        },
+        "ExecuteToolRequest": {
+          "type": "object",
+          "properties": {
+            "execution_id": {
+              "type": "string",
+              "description": "The globally-unique ID for this tool execution."
+            },
+            "trace_id": {
+              "type": "string",
+              "description": "A trace identifier for the tool execution."
+            },
+            "tool_id": {
+              "$ref": "#/components/schemas/ToolId"
+            },
+            "input": {
+              "type": "object",
+              "description": "The input parameters for the tool call.",
+              "additionalProperties": true
+            },
+            "context": {
+              "type": "object",
+              "properties": {
+                "authorization": {
+                  "type": "array",
+                  "description": "The authorization information for the tool call.",
+                  "items": {
+                    "type": "object",
+                    "properties": {
+                      "id": {
+                        "type": "string",
+                        "description": "The unique identifier for the authorization method or authorization provider."
+                      },
+                      "token": {
+                        "type": "string",
+                        "description": "The token for the tool call."
+                      }
+                    },
+                    "required": ["id", "token"],
+                    "additionalProperties": false
+                  }
+                },
+                "secrets": {
+                  "type": "array",
+                  "description": "The secrets for the tool call.",
+                  "items": {
+                    "$ref": "#/components/schemas/ToolSecret"
+                  },
+                  "additionalProperties": true
+                },
+                "user_id": {
```

Comment:
How is the validity of the user-id established? How is it used downstream?

Reply:
Specifying a valid user ID is up to the client. The downstream tool may or may not need it.
I think this would be a good thing to call out in the `requirements` section, actually -- the difference between a tool that doesn't care who the user is (`Calculator.Add`) and one that does (`Gmail.SearchEmails`)